include/ginkgo/core/base/lin_op.hpp

```diff
@@ -828,6 +828,31 @@ public:                                                                      \
                   "semi-colon warnings")
 
 
+/**
+ * Defines setters and getters for the parameters of a class, simplifying its
+ * construction by removing the repetitive typing of boiler plate code.
```

Comment:
I would add some description about accessing the variable `_concrete_parameter_name_` and the class can not use these parameter in the initial/generate step